CodeEdit/Welcome/WelcomeView.swift

```diff
@@ -43,6 +43,27 @@ struct WelcomeView: View {
         return Bundle.main.object(forInfoDictionaryKey: "CFBundleVersion") as? String ?? ""
     }
 
+    private var macOsVersion: String {
+        let osVersion = ProcessInfo.processInfo.operatingSystemVersion
+        return "\(osVersion.majorVersion).\(osVersion.minorVersion).\(osVersion.patchVersion)"
+    }
+
+    private var xCodeVersion: String {
+        guard let url = NSWorkspace.shared.urlForApplication(withBundleIdentifier: "com.apple.dt.Xcode"),
+              let bundle = Bundle(url: url) else {
+            print("Xcode is not installed")
+            return ""
```

Comment:
Maybe return, no Xcode found.

Reply:
@wdg : `Xcode not found.` o `Not found.`?

Reply:
I have some examples.

1) **Get only version number**
```swift
    private var xcodeVersion: String? {
        guard let url = NSWorkspace.shared.urlForApplication(withBundleIdentifier: "com.apple.dt.Xcode"),
              let bundle = Bundle(url: url),
              let infoDict = bundle.infoDictionary,
              let version = infoDict["CFBundleShortVersionString"] as? String
        else {
            return nil
        }

        return "\(version) (\(build))"
    }
```

2) **Get version and build number**
The displayed version number is hiding in another plist called `version.plist`.
```swift
    private var xcodeVersion: String? {
        guard let url = NSWorkspace.shared.urlForApplication(withBundleIdentifier: "com.apple.dt.Xcode"),
              let bundle = Bundle(url: url),
              let infoDict = bundle.infoDictionary,
              let version = infoDict["CFBundleShortVersionString"] as? String,
              let buildURL = URL(string: "\(url)Contents/version.plist"),
              let buildDict = try? NSDictionary(contentsOf: buildURL, error: ()),
              let build = buildDict["ProductBuildVersion"]
        else {
            return nil
        }

        return "\(version) (\(build))"
    }
```

x) Optional: **Also get MacOS Build number**
This one is a little bit weird.
There is no native swift way to get the OS build.
`ProcessInfo.processInfo.operatingSystemVersionString` prints on my (dutch) machine: `Versie 12.3 (Build 21E230)`
To fix this to `12.3 (21E230)`, I split the string, remove `Versie`(at: 0) and remove `(Build` (at: 1), and merge them all together again.
```swift
    private var macOsVersion: String {
        var osSeperated = ProcessInfo.processInfo.operatingSystemVersionString.components(separatedBy: " ")
        if osSeperated.count > 1 {
            osSeperated.remove(at: 0) // localized string (Version, Versie)
            osSeperated.remove(at: 1) // (Build
        }

        return "\(osSeperated[0]) (\(osSeperated[1])"
    }
```

**For all above examples use this, to hide the field Xcode if Xcode isn't installed on the system**
```swift
                    .onTapGesture {
                        var copyString = "CodeEdit: \(appVersion) (\(appBuild))\n" +
                        "MacOS: \(macOsVersion)\n"

                        if let xcodeVersion = xcodeVersion {
                            copyString.append("Xcode: \(xcodeVersion)")
                        }

                        let pasteboard = NSPasteboard.general
                        pasteboard.clearContents()
                        pasteboard.setString(copyString, forType: .string)
                    }
```

Reply:
@wdg : If the xCode was installed in a different place, what happens?

Reply:
It will read from that directory.

Reply:
@Angelk90 do you have discord?

Reply:
@wdg : You can do a pull on my report, so I see exactly how you want it, I take a look at it and then I do the merge.